scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1059,6 +1059,8 @@ def run(self):
         sys.stdout.flush()
 
     def gather_metrics(self, instance: TestInstance):
+        if self.options.create_rom_ram_report:
+            self.run_build(['--build', self.build_dir, "--target", "footprint"])
```

Comment:
this is calling the build twice, why not just call the main build with the target instead, something along the lines of:

```patch
--- a/scripts/pylib/twister/twisterlib/runner.py
+++ b/scripts/pylib/twister/twisterlib/runner.py
@@ -1029,7 +1029,10 @@ class ProjectBuilder(FilterBuilder):
         return res

     def build(self):
-        res = self.run_build(['--build', self.build_dir])
+        if self.options.create_rom_ram_report:
+            res = self.run_build(['--build', self.build_dir, "--target", "footprint"])
+        else:
+            res = self.run_build(['--build', self.build_dir])
         return res


```

Reply:
Thanks for the review. I'm not a Ninja expert, but I assume building the "footprint" target means the default target is not built, so it's not guaranteed to be the same thing I think. Calling build twice doesn't redo the whole build (I assume), whatever's already been built will be skipped when doing the footprint target.